operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ExecutorServiceManager.java

```diff
@@ -68,20 +71,51 @@ public ExecutorService workflowExecutorService() {
     return workflowExecutor;
   }
 
+  /**
+   * Runs the specified I/O-bound task and waits for its completion using the new ExecutorService
+   *
+   * @param task task to run concurrently
+   * @param threadNamePrefix the prefix with which to prefix thread names when tasks are run this
+   *        way
+   */
+  public static void executeAndWaitForCompletion(Runnable task, String threadNamePrefix) {
+    final var executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+    ExecutorService instrumented = new InstrumentedExecutorService(executor);
+    // change thread name for easier debugging
+    final var thread = Thread.currentThread();
+    final var name = thread.getName();
+    try {
+      thread.setName(threadNamePrefix + "-" + thread.getId());
```

Comment:
Why is the thread name changes here? It's not propagated to the executor, and basically there is just submits tasks and shutdows the executor.